shared/js/background/https.es6.js

```diff
@@ -1,151 +1,60 @@
+    setLists (lists) {
+        lists.forEach((list) => {
+            if (!list.data) return
+
+            if (list.type === 'upgrade list') {
+                let name = list.name
+                this.upgradeLists[name] = this.createBloomFilter(list.data)
+            } else if (list.type === 'whitelist') {
+                this.whitelist = list.data
             }
         })
-    }
 
-    getFromStorage (fn) {
-        return new Promise((resolve, reject) => {
-            // For Chrome/Firefox:
-            if (window.chrome) {
-                console.log('HTTPS: getFromStorage() using chrome.storage.local (Chrome/FF)')
-
-                chrome.storage.local.get('https-upgrade-list', (results) => {
-                    if (!results || !results['https-upgrade-list']) {
-                        return reject(new Error('couldn\'t get HTTPS results from local storage'))
-                    }
-
-                    try {
-                        let parsedList = JSON.parse(results['https-upgrade-list'])
-                        resolve(parsedList)
-                    } catch (e) {
-                        console.log('HTTPS: getFromStorage() error parsing JSON from chrome.storage.local', e)
-                        reject(e)
-                    }
-                })
-            } else {
-                reject(new Error('no way to get HTTPS results from local storage'))
-            }
-        })
+        this.isReady = true
     }
 
-    saveToStorage (data) {
-        // For Chrome/FF:
-        if (window.chrome) {
-            console.log('HTTPS: saveToStorage() using chrome.storage.local (Chrome/FF)')
-
-            chrome.storage.local.set({ 'https-upgrade-list': data })
-
-        // For Safari:
-        } else if (window.localStorage) {
-            console.log('HTTPS: saveToStorage() using localStorage (Safari)')
-
-            // See comment at top of the file. Need to chunk it up for Safari/localStorage
-            // because individual items in localStorage seem to have a 2.5MB limit.
-            const chunkSize = LOCAL_STORAGE_MAX_ITEM_LENGTH
-            const numChunks = Math.ceil(data.length / chunkSize)
-            for (let i = 0; i < numChunks; i++) {
-                localStorage['https-upgrade-list' + i] = data.substr(i * chunkSize, chunkSize)
-            }
-        }
+    // create a new BloomFilter
+    // filterData is assumed to be base64 encoded 8 bit typed array
+    createBloomFilter (filterData) {
+        let bloom = new BloomFilter(filterData.totalEntries, filterData.errorRate)
+        bloom.importData(Buffer.from(filterData.bloomFilter, 'base64'))
+        return bloom
     }
 
     canUpgradeHost (host) {
-        return (httpsUpgradeList.indexOf(host) > -1)
-    }
+        if (this.whitelist.indexOf(host) !== -1) {
+            return false
+        }
 
-    getUpgradeList () {
-        return httpsUpgradeList
+        let canUpgrade = false
+        Object.keys(this.upgradeLists).some((listName) => {
```

Comment:
We only have one upgrade list for now but this should handle multiple lists when/if we get there.